components/file-upload/file-uploader.ts

```diff
@@ -1,73 +1,119 @@
 }
 
 export class FileUploader {
-  public url:string;
-  public authToken:string;
-  public isUploading:boolean = false;
-  public queue:Array<any> = [];
-  public progress:number = 0;
-  public autoUpload:boolean = false;
-  public isHTML5:boolean = true;
-  public removeAfterUpload:boolean = false;
-  public queueLimit:number;
+
+  public authToken: string;
+  public isUploading: boolean = false;
+  public queue: Array<any> = [];
+  public progress: number = 0;
   public _nextIndex = 0;
-  public filters:Array<any> = [];
-  private _failFilterIndex:number;
+  private _failFilterIndex: number;
 
-  constructor(public options:any) {
-    // Object.assign(this, options);
-    this.url = options.url;
-    this.authToken = options.authToken;
-    this.filters.unshift({name: 'queueLimit', fn: this._queueLimitFilter});
-    this.filters.unshift({name: 'folder', fn: this._folderFilter});
-  }
+  public options: FileUploaderOptionsInterface = {
+    autoUpload: false,
+    isHTML5: true,
+    filters: [],
+    removeAfterUpload: false,
+  };
 
-  public addToQueue(files:any[], options:any, filters:any) {
-    let list:any[] = [];
-    for (let file of files) {
-      list.push(file);
-    }
+  @Output() events$: EventEmitter<any> = new EventEmitter();
 
-    let arrayOfFilters = this._getFilters(filters);
-    let count = this.queue.length;
-    let addedFileItems:any[] = [];
+  constructor(@Inject(Http) private http: Http) {
+  }
 
-    list.map(some => {
-      let temp = new FileLikeObject(some);
+  public setOptions(options: any) {
+    this.options = Object.assign(this.options, options);
 
-      if (this._isValidFile(temp, arrayOfFilters, options)) {
-        let fileItem = new FileItem(this, some, options);
-        addedFileItems.push(fileItem);
-        this.queue.push(fileItem);
-        this._onAfterAddingFile(fileItem);
-      } else {
-        let filter = arrayOfFilters[this._failFilterIndex];
-        this._onWhenAddingFileFailed(temp, filter, options);
-      }
-    });
+    this.authToken = options.authToken;
+    this.options.filters.unshift({name: 'queueLimit', fn: this._queueLimitFilter});
 
-    if (this.queue.length !== count) {
-      this._onAfterAddingAll(addedFileItems);
-      this.progress = this._getTotalProgress();
+    if (this.options.maxFileSize) {
+      this.options.filters.unshift({name: 'fileSize', fn: this._fileSizeFilter});
     }
 
-    this._render();
+    if (this.options.allowedFileType) {
+      this.options.filters.unshift({name: 'fileType', fn: this._fileTypeFilter});
+    }
 
-    if (this.autoUpload) {
-      this.uploadAll();
+    if (this.options.allowedMimeType) {
+      this.options.filters.unshift({name: 'mimeType', fn: this._mimeTypeFilter});
     }
+
+    // this.options.filters.unshift({name: 'folder', fn: this._folderFilter});
   }
 
-  public removeFromQueue(value:any) {
+
+  public addToQueue(files: any[], options?: any, filters?: any) {
+    this._removeFoldersFromFiles(files)
```

Comment:
@valorkin this is missing in master ... oh, no. ... Ok I fork master, create new branch(s) and lot of PRs